pkg/certificate/manager.go

```diff
@@ -178,13 +188,18 @@ func (m *Manager) nextRotationDeadline() time.Time {
 		return m.now()
 	}
 
-	caCert, err := m.getFirstCACertFromCABundle()
+	// Last rotated CA cert at CABundle is the last at the slice so this
+	// calculate deadline from it.
+	caCert, err := m.getLastCACertFromCABundle()
```

Comment:
I still don't get why we take the last. 
my reasoning is: last caBundle is the last rotated, so it's expiration will be the furthest. don't we want the one with the earliest expiration?

Reply:
This is the rotation deadline not the cleanup deadline.

Reply:
ok, but why the last?

Reply:
Because we `append` the rotated cert so the last certificate is the one rotated.

Reply:
ok I understand. 
Then since you are relying on the functionality of `append` function, maybe consider to change the function name to `getLastAppendedCACertFromCABundle`. I know it's long but I personally prefer it to be more readable like so.